### PR TITLE
Adding validation for email when verifyEmail claim is set

### PR DESCRIPTION
**Pull request by ThaminduDilshan**
### Proposed changes in this pull request

If a curl request is made (ask password functionality is enabled), it returns a 500 response along with a stack trace. But the user is already created and it returns 409 if the same request is made.

The issue is that "verifyEmail" is set as true, it is trying to send a verification mail on POST_ADD_USER since an email is not provided for the user. The user is already created, but SCIM returns 500.

Fix is to add a validation step on PRE_ADD_USER.

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

**Related issue**
- https://github.com/wso2/product-is/issues/13637

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java [hunk 1]
```diff
@@ -378,7 +378,11 @@ public enum ErrorMessages {
         ERROR_CODE_INVALID_ANSWER_FORMAT("10016", "Invalid answer format in the given answer " +
                 "for the challenge question '%s'."),
         ERROR_CODE_NOT_UNIQUE_ANSWER("10017", "The given answer for the challenge question, " +
-                "'%s' has been used more than once.");
+                "'%s' has been used more than once."),
+
+        // UEV - User Email Verification.
+        ERROR_CODE_VERIFY_EMAIL_NOT_SET("UEV-10001", "verifyEmail claim is set, but the email addresses are not " +
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java:384]: `verifyEmail claim is set, but the email addresses are not provided` > `Email address not found for email verification` WDYT?

Comment (somindatommy) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java:384]: `verifyEmail claim is set, but the email addresses are not provided` > `Email address not found for email verification` WDYT?

**Review by somindatommy — APPROVED** [review]

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java (hunk 1 above)
Comment (nilasini) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java:384]: The error code should imply that email address not found right? Shall we change the constant from `ERROR_CODE_VERIFY_EMAIL_NOT_SET to ERROR_CODE_EMAIL_ADDRESS_NOT_FOUND`

Comment (ThaminduDilshan) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java]: Fixed

Comment (ThaminduDilshan) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/IdentityRecoveryConstants.java]: Updated as for the offline chat

**Review by nilasini — APPROVED** [review]

**Pull request merged**